presto-hive/src/main/java/com/facebook/presto/hive/HiveFileWriter.java

```diff
@@ -25,7 +28,7 @@
 
     void appendRows(Page dataPage);
 
-    void commit();
+    List<FileColumnStatistics> commit();
```

Comment:
The return format should be a page (or maybe a list of pages). Or if you wanna wrap it, it is also ok. But `Page` is the essential part for it. That allows the coordinator to write pages into manifests in ORC format.

Here is an example: 

| file_prefix | file_name | file_suffix | rows | bytes | col1_min | col1_max | col1_non_null_rows | col2_min | col2_max | col2_non_null_rows |
| --  | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
| ws://ws.atn5/abc/ | 1 | .orc | 1000 | 100030 | ab | zzzd | 900 | 0 | 15 | 1000 |
| ws://ws.atn5/abc/ | 2 | .orc | 1200 | 110040 | ac | zzf | 1100 | 5 | 22 | 1200 |
| ws://ws.atn5/abc/ | 3 | .orc | 1100 | 120050 | bf | xwf | 1000 | 19 | 42 | 1100 |


Each column types should be known beforehand. Or if not, types are serializable anyway, so they can be saved as `List<TypeSignature>`. For example, the above stats should come with column types: {varchar, varchar, varchar, bigint, bigint, varchar, varchar, bigint, bigint, integer, integer}.

Types and Blocks are the core data structure for Presto. So let's try to reuse them as much as possible.